py-rattler/rattler/package/index_json.py

```diff
@@ -296,12 +297,12 @@ def timestamp(self) -> Optional[int]:
         ...     "../test-data/with-symlinks/python-3.10.6-h2c4edbf_0_cpython.tar.bz2"
         ... )
         >>> idx_json.timestamp
-        1661200742
+        datetime.datetime(2022, 8, 22, 22, 39, 2, 467000)
         >>>
         ```
         """
         if timestamp := self._inner.timestamp:
-            return timestamp
+            return datetime.datetime.fromtimestamp(timestamp / 1000.0)
```

Comment:
I'm wondering if we could potentially do something like
```python
datetime.datetime(1970, 1, 1, 0, 0, 0, 0, datetime.timezone.utc) + datetime.timedelta(milliseconds=timestamp)
```
to keep the precision. Also note that I think we should return an aware datetime instead of a naive datetime.

Reply:
I was under the impression that this timestamp: `        datetime.datetime(2022, 8, 22, 22, 39, 2, 467000)` contains milliseconds. But also happy to change it.

Reply:
Ah, I see that you are keeping it as a float, so yeah then that works. In this case, you can use `datetime.datetime.fromtimestamp()` and you can pass a timezone
```python
datetime.datetime.fromtimestamp(timestamp / 1000.0, tz=datetime.timezone.utc)
```

Reply:
will add the `utc` parameter.